src/components/controller/units/UnitsTable.vue

```diff
@@ -63,16 +68,39 @@ onMounted(() => {
   hideOpenUnitPopupsTooltip.value = getPreference('hideOpenUnitPopupsTooltip', loginStore.username)
 })
 
-async function openUnit(unitId: string) {
+async function openUnit(unit: Unit, versionCheck = true) {
   error.value.openUnit = ''
+  currentUnit.value = unit
+  loadingOpenUnit.value = true
 
   try {
-    await unitsStore.checkUnitToken(unitId)
-    const routeData = router.resolve({ path: `/controller/manage/${unitId}/dashboard` })
-    window.open(routeData.href, '_blank')
+    await unitsStore.getUnitInfo(currentUnit.value.id)
+    await unitsStore.getUnits()
+    // Find the now updated unit, as the currentUnit might have changed
+    currentUnit.value = unitsStore.units.find((u) => u.id == unit.id)!
+    // Logic on which message is shown is inside the ObsoleteApiModal component
+    let version = currentUnit.value?.info.api_version
+    if (version == '') {
```

Comment:
When does this happen? Can `version` also be `null`  or `undefined`?

Reply:
It can if you're browsing a current stable unit with the latest controller, the `api_version` field is empty, while `currentUnit.value?` is `Unit | undefined` by nature